app/exec/extension/install.ts

```diff
@@ -0,0 +1,279 @@
+        const accountReport: AccountInstallReport = new AccountInstallReport(itemId, accountName, accountId);
+
+        if (!accountId) {
+            accountReport.setError(`Cannot install extension into account ${accountName} because is not an acccount for which the installation user is owner or a member of.`);
+            return Q.resolve(accountReport);
+        }
+
+        return galleryApi.requestAcquisition(<GalleryInterfaces.ExtensionAcquisitionRequest>{
+            assignmentType: GalleryInterfaces.AcquisitionAssignmentType.None,
+            billingId: null,
+            itemId: itemId,
+            operationType: GalleryInterfaces.AcquisitionOperationType.Install,
+            targets: [accountId]
+        })
+            .then(report => {
+                trace.debug(`Succeeded GalleryApi.requestAcquisition in ${accountName}: ${report}`);
+                accountReport.installed = true;
+                return accountReport;
+            })
+            .fail((err) => {
+                // Ignore error 'AlreadyInstalled'
+                if (this._isAlreadyInstalledError(err)) {
+                    trace.debug(`Succeeded GalleryApi.requestAcquisition in ${accountName}: Alreeady installed`);
+                    accountReport.setInstalled("Extension already installed");
+                    return accountReport;
+                }
+
+                trace.error(`Failed GalleryApi.requestAcquisition for account ${accountName}: ${err}`);
+                accountReport.setError(err);
+                return accountReport;
+            });
+    }
+
+    private _isAlreadyInstalledError(err): boolean {
+        // 409 - Conflict: means extension already installed
+        return err.statusCode === 409;
+    }
+
+    private _getAccountsByMemberId(userId: string): Q.Promise<Account[]> {
+        const accountsBaseUrl = this._getAccountsServiceBaseUrl(this.webApi.serverUrl);
+        const accountsApi = new AccountsApi(accountsBaseUrl, [this.webApi.authHandler]);
+        return accountsApi.getAccountsByMemberId(userId);
+    }
+
+
+    /**
+     * Determine the Accounts URL based on the service URL.
+     * TODO: Check if it is possible to do it using the REST API.
+     *
+     * @param  {string} serviceUrl
+     * @returns string
+     */
+    private _getAccountsServiceBaseUrl(serviceUrl: string): string {
+        // When hosted, return SPS URL
+        if (serviceUrl.toLowerCase().indexOf(".visualstudio.com") > 0) {
+            return VSTS_SPS_BASE_URL;
+        }
+
+        // Otherwise return the original serviceUrl
+        return serviceUrl;
+    }
+}
+
+
+/**
+ * TODO: Remove this as soon as it is included in the vso-node-api module
+ *
+ */
+
+import VsoBaseInterfaces = require("vso-node-api/interfaces/common/VsoBaseInterfaces");
+import { ClientApiBase } from  "vso-node-api/ClientApiBases";
+
+class AccountsApi extends ClientApiBase {
```

Comment:
@trevorsg While AccountsApi is not included vso-node-api this is the simplest way to get the list of accounts by member id.